File: src/main/java/com/group/libraryapp/domain/user/UserRepository.java
package com.group.libraryapp.domain.user;

import com.group.libraryapp.domain.user.loan_history.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {

  Optional<User> findByName(String name);

}

File: src/main/kotlin/com/group/libraryapp/domain/user/UserRepository.kt
package com.group.libraryapp.domain.user

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface UserRepository: JpaRepository<User,Long> {

/*
    자바
    Optional<User> findByName(String name);
*/

    fun findByName(name: String) : User?

    @Query("select distinct u from User u left join fetch u.userLoanHistories")
    fun findAllWithHistories(): List<User>

}